lib/PeactDOM.js

```diff
@@ -0,0 +1,114 @@
+const prevSnapshot = { vDOM: null };
+
+/* initial rendering 함수 */
+function render(vDOM, root) {
+  // parse => createElement(...)
+  prevSnapshot.vDOM = vDOM;
+
+  const fragment = document.createDocumentFragment();
+  makeDOM(vDOM, fragment);
+  root.appendChild(fragment);
+}
+
+/* Virtual DOM을 순회하며 element를 생성하는 함수 */
+function makeDOM(vDOMObject, parent) {
+  const { type, props, children } = vDOMObject;
+
+  if (type === 'text') {
+    parent.textContent = children[0];
+    return;
+  }
+
+  const element = document.createElement(type);
+
+  Object.entries(props).forEach(([key, value]) =>
+    typeof value === 'function'
+      ? element.addEventListener(key, value)
+      : element.setAttribute(key, value)
+  );
+
+  parent.appendChild(element);
+
+  children.forEach((child) => {
+    makeDOM(child, element);
+  });
+}
+
+/* re-rendering 함수 */
+function reRender(newVDOM) {
+  const rootElement = document.getElementById('root');
+  const appComponent = rootElement.children[0];
```

Comment:
```suggestion
  const rootElement = document.getElementById('root');
  const appComponent = rootElement.firstChild;
```
첫 번째 자식 요소를 얻고 싶다면 조금 더 선언적으로 얻는 방법도 있을 것 같아용 :)

Reply:
오 좋습니다!!